packages/core/src/client/hmr/overlay.ts

```diff
@@ -165,12 +165,25 @@ if (customElements && !customElements.get(overlayId)) {
   customElements.define(overlayId, ErrorOverlay);
 }
 
+const documentAvailable = typeof document !== 'undefined';
+
 export function createOverlay(err: string[]) {
+  if (!documentAvailable) {
+    console.info(
+      'Failed to display Rsbuild overlay since document is not available, considering turning off the overlay option.',
```

Comment:
```suggestion
      'Failed to display Rsbuild overlay since document is not available, considering turning off the `dev.client.overlay` option.',
```

Reply:
> ```suggestion
> 
>       'Failed to display Rsbuild overlay since document is not available, considering turning off the `dev.client.overlay` option.',
> 
> ```

user may encounter this info in upper framework, the config then will be varied.

Reply:
Yeah, but currently all known upper frameworks retain the original `dev.client` configuration as Rsbuild. Vite will also display the of original config name, so I tend to display `dev.client.overlay` in the log.

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/c2db3602-0976-4f68-b4b1-a12c6029f5fb)